### PR TITLE
fix(repos): squash duped repos

### PR DESCRIPTION
**Pull request by cathteng**
We removed a constraint on the repository table in #53286, but this has been causing more issues with duplicates than before. This PR squashes duped repos in preparation for putting back the constraint on the table for org_id + name. We should then ensure that we update the repo name if it changes but the external id is the same.

### CODE REVIEW

File: src/sentry/migrations/0553_squash_same_name_repos.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def squash_duped_repos(apps, schema_editor):
+    Repository = apps.get_model("sentry", "Repository")
+    PullRequest = apps.get_model("sentry", "PullRequest")
+    ReleaseHeadCommit = apps.get_model("sentry", "ReleaseHeadCommit")
+    LatestRepoReleaseEnvironment = apps.get_model("sentry", "LatestRepoReleaseEnvironment")
+    Commit = apps.get_model("sentry", "Commit")
+
+    duplicates = Repository.objects.values("name").annotate(count=Count("name")).filter(count__gt=1)
```
Comment (scefali) [inline src/sentry/migrations/0553_squash_same_name_repos.py:16]: @cathteng we probably need to use the ranged query set thing

Comment (scefali) [inline src/sentry/migrations/0553_squash_same_name_repos.py:16]: @cathteng any idea of the performance of this query? We'll be calling it many times. I'm concerned it's going to be slow. Might be safer to do this manually by iterating through all the repos and for each one find a row of the same organization_id, name, but different id.

Comment (cathteng) [reply src/sentry/migrations/0553_squash_same_name_repos.py:16]: i ran the equivalent in redash and it returned pretty quickly. there are 12789 duplicates as i am posting this comment

Comment (cathteng) [reply src/sentry/migrations/0553_squash_same_name_repos.py:16]: when you do `values("name")`, you just get a list of repo names that have duplicates, not a `BaseQueryset`

Comment (scefali) [reply src/sentry/migrations/0553_squash_same_name_repos.py:16]: @cathteng That's going to be generating 12k of them in memory. Might be Ok. Also, shouldn't you check if the org id is the same? It's ok for 2 names to be the same on different orgs.

Comment (cathteng) [reply src/sentry/migrations/0553_squash_same_name_repos.py:16]: oh yeah

Comment (cathteng) [reply src/sentry/migrations/0553_squash_same_name_repos.py:16]: oh jk this is still a BaseQuerySet so i can use RangeQuerySetWrapper

**Pull request closed without merging**